test/shared_utils/utilities.py

```diff
@@ -100,9 +104,14 @@ def compare_results_to_reference_file_new(
     reference_data = DataTree.from_dict(reference_groups)
     results_data = DataTree.from_dict(results_groups)
 
-    assert results_data.identical(reference_data), (
-        'Output and reference files ' 'do not match.'
-    )
+    if identical:
+        assert results_data.identical(
+            reference_data
+        ), 'Output and reference files do not match.'
+    else:
+        assert results_data.equals(
```

Comment:
Yes, this is due to Trajectory Subsetter adding a `Processing Parameters` field 
this global attribute begins with:

``` shell
:Processing\ Parameters = "This file was generated by the ICESat-2 Service version 1.0, which performed the following operations on 456eaac9b2e3b9c7311ab68d6f7d479ff2179bdfb01cfef374d1cf5849b6cef3.h5
``` 

Where the filename is a hash of the full path to the granule from harmony-service-lib:
https://github.com/nasa/harmony-service-lib-py/blob/main/harmony_service_lib/util.py#L273-L276


I have written a ticket to allow the comparison to ignore attributes by name, but thought that was just a bit more than I wanted to jam into this fire drill. There could be discussion if the filename shouldn't be a hash at all, but that's a different Matt's problem.


(as an aside: you can see that the metadata (name and version?) is also incorrect because the version is hard coded: 
https://git.earthdata.nasa.gov/projects/SITC/repos/trajectorysubsetter/browse/subsetter/Subsetter.h#766-769 I did not write a ticket for this)

Reply:
re 2.  "There could be discussion if the filename shouldn't be a hash at all, but that's a different Matt's problem." 

It's certainly worth a discussion of all of those alternatives.

Reply:
So, fun fact:

[This](https://github.com/nasa/harmony-service-lib-py/blob/main/harmony_service_lib/util.py#L281-L282) would suggest that the intended file name _should_ be the basename from the URL. [But it's clearly not](https://github.com/nasa/harmony-service-lib-py/blob/main/harmony_service_lib/util.py#L275). It's probably a case of a stale documentation string (it's 2 years older than the function that creates the file name), so something to chat about with the Harmony folks.